### ISSUE TITLE
confignetwork fails when BMC interfaces are defined in the 'nics' table

### ISSUE THREAD

**Issue by kcgthb** [xcat2/xcat-core#4538]
The `confignetwork` postscript returns an error when BMC interfaces are defined in the 'nics` table.

For instance, with  the following node definition:
```
Object name: sh-06-34
    nicaliases.bmc=sh-06-34
    nicaliases.ib0=sh-06-34
    nicextraparams.ib0=MTU=65520 CONNECTED_MODE=yes
    nicips.bmc=10.11.06.34
    nicips.ib0=10.8.06.34
    nicnetworks.bmc=net_ipmi
    nicnetworks.ib0=net_ib_56
    nictypes.bmc=bmc
    nictypes.ib0=infiniband
```

```
# updatenode sh-06-34 -P confignetwork
sh-06-34: xcatdsklspost: downloaded postscripts successfully
sh-06-34: Fri Dec 15 09:23:49 PST 2017 Running postscript: confignetwork
sh-06-34: [I]: All valid nics and device list:
sh-06-34: [I]: bmc
sh-06-34: [I]: ib0
sh-06-34: [I]: NetworkManager is inactive.
sh-06-34: ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
sh-06-34: configure nic and its device : bmc
sh-06-34: [E]:Error: Check the NIC data in the 'nics' table.
sh-06-34: ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
sh-06-34: configure nic and its device : ib0
sh-06-34: [I]: Call configib for IB nics: ib0, ports:
sh-06-34: [I]: NIC_IBNICS=ib0 NIC_IBAPORTS= configib
sh-06-34: em1!MTU=9000
sh-06-34: ib0!MTU=65520 CONNECTED_MODE=yes
sh-06-34: array_nic_params 0=MTU=65520 CONNECTED_MODE=yes
sh-06-34:   0: name=MTU value=65520
sh-06-34:   1: name=CONNECTED_MODE value=yes
sh-06-34: postscript: confignetwork exited with code 1
sh-06-34: Running of postscripts has completed.
```

The script produces an error for the `bmc` interface:
```
[E]:Error: Check the NIC data in the 'nics' table.
```
which kind of makes sense, since the OS doesn't really see the BMC interface and can't configure it. Yet `bmc` is a valid interface type in the `nics` table, as mentioned in the [documentation](https://xcat-docs.readthedocs.io/en/stable/advanced/domain_name_resolution/domain_name_resolution.html?highlight=bmc#specifying-additional-network-interfaces-for-cluster-nodes). So the script should not exit with an error.

We need a way to define IP <-> hostname resolution for our BMC interfaces, so they get generated by `makehosts` and `makedns`.

So, what's the way to define hostname and IP addresses for BMC interfaces now?

**Comment by kcgthb** [comment]
PR in #4539

**Comment by mattaezell** [comment]
Hi @kcgthb.  We use [`otherinterfaces`](http://xcat-docs.readthedocs.io/en/stable/guides/admin-guides/references/man5/hosts.5.html?highlight=otherinterfaces).  It ties in with `makehosts`, but I'm not sure about `makedns`

**Comment by bybai** [comment]
Hi @kcgthb, We recommend to use ``otherinterfaces`` in ``hosts`` table. Please refer to doc http://xcat-docs.readthedocs.io/en/stable/QA/makehosts.html#q-how-to-configure-aliases , thanks.

**Comment by kcgthb** [comment]
Well, that's what I used to do in earlier versions of xCAT, until the introduction of the `nics` table and the `confignetwork` postscript, which deprecated `otherinterfaces`.

For a time, there was a mention of the deprecation in the [Domain Name Resolution](https://xcat-docs.readthedocs.io/en/stable/advanced/domain_name_resolution/domain_name_resolution.html) page of the documentation:

> For now the “otherinterfaces” attribute will still be supported but it may be dropped in future releases.Any new network interface information should be provided using the new nic attributes.

That mention has been recently removed (in commit 29ff06b) with the following message:
> Remove otherinterfaces which has been replaced by the nic attributes starting in xCAT 2.8.1

I'll go back to using `otherinterfaces` now, but it's all **very** confusing. It would be a good thing to update the documentation to state things more clearly: what is deprecated, what is not, how should users define IP addresses for interfaces that `confignetwork` cannot set up.

**Comment by kcgthb** [comment]
Well, I gave `otherinterfaces` a try, but it doesn't fit my use case: I need to have the same hostname defined for a node and its BMC interface in 2 separate domaines. Like `cn.local` and `cn.ipmi`. Turns out I can do this with `nics`+`confignetwork`, but not with `otherinterfaces` as those other interfaces require a suffix. 
I've seen https://xcat-docs.readthedocs.io/en/stable/QA/makehosts.html#q-how-to-handle-the-same-short-hostname-in-different-domains but I don't want to define 2 objets in the DB for every single node.

So I'm back with my initial solution of defining the BMC IP address in `nics` and ignoring interfaces of type `bmc` in `confignetwork`. I don't think it would hurt to skip those NIC types in the `confignetwork` script anyway.
